### PR TITLE
Endpoint tests and implementation [12833]

### PR DESCRIPTION
**Pull request by jparisu**
Merge after #26 & #27 

Signed-off-by: jparisu <javierparis@eprosima.com>

### CODE REVIEW

File: test/unittest/types/topic/CMakeLists.txt [hunk 1]
```diff
@@ -39,6 +39,8 @@ set(TEST_LIST
 set(TEST_EXTRA_LIBRARIES
     )
 
+set(TEST_EXTRA_LIBRARIES)
+
```
Comment (rsanchez15) [inline test/unittest/types/topic/CMakeLists.txt:43]: Duplicated after rebase

**Review by rsanchez15 — CHANGES_REQUESTED** [review]
- [x] In `DataReceived` change `QoS` by `Guid`

File: test/unittest/types/topic/CMakeLists.txt [hunk 2]
```diff
@@ -110,6 +112,8 @@ set(TEST_EXTRA_LIBRARIES
         $<$<BOOL:${WIN32}>:iphlpapi$<SEMICOLON>Shlwapi>
     )
 
+set(TEST_EXTRA_LIBRARIES)
```
Comment (rsanchez15) [inline test/unittest/types/topic/CMakeLists.txt:115]: Removing this will fix the windows compilation error

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

File: test/unittest/types/topic/CMakeLists.txt (hunk 1 above)
Comment (jparisu) [reply test/unittest/types/topic/CMakeLists.txt:43]: :+1:

File: test/unittest/types/topic/CMakeLists.txt (hunk 2 above)
Comment (jparisu) [reply test/unittest/types/topic/CMakeLists.txt:115]: :+1:

**Review by rsanchez15 — APPROVED** [review]

**Pull request merged**